.github/workflows/integration.yaml

```diff
@@ -0,0 +1,105 @@
+name: 'Integration'
+
+on:
+  push:
+    branches:
+    - 'main'
+  pull_request:
+    branches:
+    - 'main'
+  workflow_dispatch:
+
+concurrency:
+  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+permissions:
+  contents: 'read'
+  id-token: 'write'
+
+env:
+  REGION: us-central1
+
+jobs:
+  test:
+    if: ${{ github.event_name == 'push' || github.repository == github.event.pull_request.head.repo.full_name && github.actor != 'dependabot[bot]' }}
+    runs-on: 'ubuntu-latest'
+
+    steps:
+    - id: 'checkout'
+      name: 'Checkout'
+      uses: 'actions/checkout@v3'
+
+    - id: 'compute-test-uid'
+      name: 'Compute test UID'
+      run: |-
+        echo "TEST_UID=${GITHUB_JOB}-${GITHUB_SHA::7}-${GITHUB_RUN_NUMBER}" >> ${GITHUB_ENV}
+
+    - id: 'set-up-node'
+      name: 'Set up Node'
+      uses: 'actions/setup-node@v3'
+      with:
+        node-version: '16.x'
+
+    - id: 'build'
+      name: 'Build'
+      run: 'npm ci && npm run build'
+
+    - id: 'auth'
+      name: 'Auth'
+      uses: 'google-github-actions/auth@v1'
+      with:
+        workload_identity_provider: '${{ secrets.WIF_PROVIDER_NAME }}'
+        service_account: '${{ secrets.SERVICE_ACCOUNT_EMAIL }}'
+
+    - id: 'gcloud-setup'
+      name: 'Set up Cloud SDK'
+      uses: google-github-actions/setup-gcloud@v0
+      with:
+        project_id: ${{ secrets.PROJECT_ID }}
+
+    - id: 'populate-pipeline-template'
+      name: 'Populate values in template for Cloud Deploy pipeline and target'
+      run: 'envsubst < tests/fixtures/clouddeploy.yaml.template > tests/fixtures/clouddeploy.yaml'
+      env:
+        PROJECT_ID: '${{ secrets.PROJECT_ID }}'
+        REGION: '${{ env.REGION }}'
+
+    - id: 'create-pipeline'
+      name: 'Create pipeline'
+      run: 'gcloud deploy apply --file=tests/fixtures/clouddeploy.yaml --region=${REGION}'
+
+    - id: 'create-release'
+      name: 'Create release'
+      uses: './'
+      with:
+        name: '${{ env.TEST_UID }}'
+        region: '${{ env.REGION }}'
+        delivery_pipeline: 'gha-test-pipeline'
```

Comment:
Where does this come from?

Reply:
The pipeline is created in the [`create-pipeline` step](https://github.com/henrybell/create-cloud-deploy-release/blob/e9aef7f1fecfa11128f150b925cc2f55e142b369/.github/workflows/integration.yaml#L68) above, from [YAML in the test fixtures](https://github.com/henrybell/create-cloud-deploy-release/blob/henrybell-import/tests/fixtures/clouddeploy.yaml.template). It's declarative and applied each time idempotently -- the pipeline isn't deleted after each test run as there may be queued tests that will use the same pipeline. Initially I was creating a new pipeline for each test run, but reuse is beneficial from a cost standpoint, as billing is on a [per-pipeline basis](https://cloud.google.com/deploy/pricing).

Reply:
How challenging is it for us to create and delete pipelines on each CI run? How much time does that add?

Depending on this "magic string", we limit parallelization options and further limit our ability to test this in other projects, since it requires a special setup in the target environment.

Reply:
The time to create or delete a pipeline is a matter of a few seconds -- it's more the fact that each pipeline incurs non-trivial cost, so if each test involves creating a pipeline, performing a release, and then deleting the pipeline, this can get expensive if many tests are being run (as the billing model is the number of active pipelines per month). The first test run in any project will create the pipeline necessary for the test, so it's self-provisioning in that regard.

Reply:
When we move to a centrally billed test project (as opposed to my sandbox), creation and subsequent deletion of a pipeline per test run will be straightforward to implement.

Reply:
Can we work on the central test project? What do you need and we can get it setup for you.